File: platform_integration/formula_api_client.ts
// TypeScript client for KURGIN Formula API.
// Intended for KURGIN Platform backend/server route usage, not direct browser calls.

export type KurginStoneInput = Record<string, unknown>;

export interface KurginFormulaClientOptions {
  baseUrl: string;
  apiKey?: string;
}

export class KurginFormulaApiClient {
  private baseUrl: string;
  private apiKey?: string;

  constructor(options: KurginFormulaClientOptions) {
    this.baseUrl = options.baseUrl.replace(/\/$/, "");
    this.apiKey = options.apiKey;
  }

  private headers(): HeadersInit {
    const headers: Record<string, string> = {
      "Content-Type": "application/json",
    };
    if (this.apiKey) headers["X-KURGIN-API-Key"] = this.apiKey;
    return headers;
  }

  async health() {
    const res = await fetch(`${this.baseUrl}/v1/health`, { cache: "no-store" });
    if (!res.ok) throw new Error(`KURGIN health failed: ${res.status}`);
    return res.json();
  }

  async analyzeStone(stone: KurginStoneInput, language = "RU") {
    const res = await fetch(`${this.baseUrl}/v1/analyze/stone`, {
      method: "POST",
      headers: this.headers(),
      body: JSON.stringify({ language, stone }),
      cache: "no-store",
    });
    if (!res.ok) throw new Error(`KURGIN analyze failed: ${res.status} ${await res.text()}`);
    return res.json();
  }
}

File: platform_integration/formula_api_client.py
"""Python client for KURGIN Formula API.

This is intended for KURGIN Platform backend usage.
"""

from __future__ import annotations

from typing import Any, Dict, Optional

import requests


class KurginFormulaApiClient:
    def __init__(self, base_url: str, api_key: str = "", timeout: int = 30):
        self.base_url = base_url.rstrip("/")
        self.api_key = api_key
        self.timeout = timeout

    def _headers(self) -> Dict[str, str]:
        headers = {"Content-Type": "application/json"}
        if self.api_key:
            headers["X-KURGIN-API-Key"] = self.api_key
        return headers

    def health(self) -> Dict[str, Any]:
        response = requests.get(f"{self.base_url}/v1/health", timeout=self.timeout)
        response.raise_for_status()
        return response.json()

    def ready(self) -> Dict[str, Any]:
        response = requests.get(f"{self.base_url}/v1/ready", timeout=self.timeout)
        response.raise_for_status()
        return response.json()

    def analyze_stone(self, stone: Dict[str, Any], language: str = "RU") -> Dict[str, Any]:
        response = requests.post(
            f"{self.base_url}/v1/analyze/stone",
            json={"language": language, "stone": stone},
            headers=self._headers(),
            timeout=self.timeout,
        )
        response.raise_for_status()
        return response.json()

    def analyze_batch_json(self, rows: list[Dict[str, Any]], language: str = "RU", include_records: bool = True):
        response = requests.post(
            f"{self.base_url}/v1/analyze/batch/json",
            json={"language": language, "rows": rows, "include_records": include_records},
            headers=self._headers(),
            timeout=self.timeout,
        )
        response.raise_for_status()
        return response.json()

    def export_stone_pdf(self, stone_result: Dict[str, Any], language: str = "BILINGUAL") -> bytes:
        response = requests.post(
            f"{self.base_url}/v1/export/stone/pdf",
            json={"language": language, "stone_result": stone_result},
            headers=self._headers(),
            timeout=self.timeout,
        )
        response.raise_for_status()
        return response.content
